### PR TITLE
Fixed product mutation when making tax calculation

### LINKED ISSUE #3740: Wrong price after first return from product page.
## Current behavior

After visiting the product list page I visit the product detail view page first time - Price there is correct. Then I return to the product list page again and then detail product view - this time I have the wrong price value (looks like it have discount of approximately 10%). this behavior seems to happen only on the second visit of the detail page, First visit and visits thereafter display the correct price.
## Expected behavior

Price should always remain the same 

## Steps to reproduce the issue

set config properties to
"sourcePriceIncludesTax": true,
"calculateServerSide": false
then
1 Load category page. 2 Load product. 3 Back to the category page.

## Can you handle fixing this bug by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [x] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser:  77.0.3865.120 Chrome
- OS: Windows 10
- Node: 12.13.0
- Code Version: VSF 1.10.4

### PR DESCRIPTION
**Pull request by gibkigonzo**
### Related issues

closes #3740 #3750

### Short description and why it's useful

It fixes price calculation when `config.tax.calculateServerSide` is set to false. Calculation can be done more then once so now we save original prices and use it for every calculation.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by pkarw** [comment]
Are you sure this solves the issue? I mean that changing the product was intended because otherwise all the product lists etc won't get the product with new prices / taxes calcualted? Have you tested if the prices are correct? Maybe we just need a flag like `product.tax_calc_ts = new Date().getTime()` in order to make sure taxes are calculated just once (?)

**Comment by gibkigonzo** [comment]
Yes, you are right. Sorry, I saw map and returns in `calculateTaxes` and this mislead me. I've added flag to product after first calculation.

File: core/modules/catalog/helpers/taxCalc.ts [hunk 1]
```diff
@@ -213,6 +213,10 @@ export function updateProductPrices ({ product, rate, sourcePriceInclTax = false
 }
 
 export function calculateProductTax ({ product, taxClasses, taxCountry = 'PL', taxRegion = '', sourcePriceInclTax = false, deprecatedPriceFieldsSupport = false, finalPriceInclTax = true, userGroupId = null, isTaxWithUserGroupIsActive }) {
+  if (product.isTaxCalculated) {
+    return product
+  }
+  product.isTaxCalculated = true
```
[the remark this answers is not in the document]
Comment (gibkigonzo) [reply core/modules/catalog/helpers/taxCalc.ts]: I've changed approach to this issue. In first calculation prices are saved with `original` in key name. Then in next calc we can use those values to safely mutate product prices. I made a little rewrite to understand this. I can make similar changes for vuestorefront-api if it's needed.

**Comment by ArturDivante** [comment]
@gibkigonzo @andrzejewsky Test instructions needed :)

**Comment by gibkigonzo** [comment]
**update**
Also fixed some bug with changing country in checkout. We didn't update country id, so it could send wrong (always the same) country in `/api/cart/shipping-information` request.

### QA:
Set in vuestorefront and vuestorefront-api config:
`calculateServerSide` to `false`
`sourcePriceIncludesTax` to `true`
`finalPriceIncludesTax` to `true`.
Anyway there should be same config on frontend and backend.

First scenario - prices should not change on category page:
1. open category page
2. open product page
3. go back to category - price should be same as in 1. step

Second scenario - tax should update in checkout: 
1. add product to checkout
2. go to shipping step
3. change country with different tax for example between Poland and Bahamas

**Comment by pkarw** [comment]
We need to update `vue-storefront-api` -> `taxCalc.js` as well @gibkigonzo please add it to your TODO

**Pull request merged**